File: src/replace_settings.mjs
import fs from 'fs';

const content = fs.readFileSync('src/App.tsx', 'utf-8');
const startStr = '{/* Settings Modal */}';
const endStr = '{/* Memory Modal Removed */}';

const startIndex = content.indexOf(startStr);
const endIndex = content.indexOf(endStr);

if (startIndex !== -1 && endIndex !== -1) {
  const newContent = content.substring(0, startIndex + startStr.length) + '\n        <SettingsModal />\n\n        ' + content.substring(endIndex);
  fs.writeFileSync('src/App.tsx', newContent);
  console.log('Successfully replaced Settings Modal');
} else {
  console.log('Could not find start or end string');
}

File: app/applet/replace_settings.py
import sys

with open('src/App.tsx', 'r') as f:
    lines = f.readlines()

start_idx = -1
end_idx = -1

for i, line in enumerate(lines):
    if '{/* Settings Modal */}' in line:
        start_idx = i
    if '{/* Memory Modal Removed */}' in line:
        end_idx = i
        break

if start_idx != -1 and end_idx != -1:
    new_lines = lines[:start_idx+1] + ['        <SettingsModal />\n\n'] + lines[end_idx:]
    with open('src/App.tsx', 'w') as f:
        f.writelines(new_lines)
    print("Successfully replaced Settings Modal")
else:
    print("Could not find start or end index")
